### DISCUSSION TITLE
Copyalpha from two diffrent images?

### DISCUSSION THREAD

**Discussion by Isak-Andersson** [ImageMagick/ImageMagick#5359] [Help] [answered]
Hello. I want to combine the alpha values from two images keeping the RGB values from the first image if that makes sense.

Edit:
I'm on Windows 10 and ImageMagick version 7.1.0.

Basically, I want to combine this:
![Input](https://user-images.githubusercontent.com/78758898/180602671-ea31dc99-36c8-4fe8-a3ec-bee42163dd32.png)
With this:
![AlphaMask](https://user-images.githubusercontent.com/78758898/180602763-b7fa01d7-44e7-4f95-91bb-d43b87be85fc.png)
To make this:
![Output](https://user-images.githubusercontent.com/78758898/180602672-09f96731-f552-4b34-888c-b71f5ebcf06b.png)

Can this be done? Thanks!

**Comment by fmw42** [comment]
What version of Imagemagick and what platform/OS?  I will assume IM 6 on Unix.

Extract the alpha channel from each and multiply together. Then put that back into the alpha channel of the input.

```
convert cross.png \
\( +clone hexagon.png -alpha extract -compose multiply -composite \) \
-alpha off -compose copy_opacity -composite \
result.png

```

![result](https://user-images.githubusercontent.com/13458218/180613291-0958276e-b5d8-4e5d-b9ff-7ece9b2453cb.png)

**Reply by Isak-Andersson** [reply]
Sorry, I forgot to mention. I'm on Windows 10 and ImageMagick version 7.1.0.

**Reply by fmw42** [reply]
```
magick.exe cross.png ^
( +clone hexagon.png -alpha extract -compose multiply -composite ) ^
-alpha off -compose copy_opacity -composite ^
result.png
```

**Reply by Isak-Andersson** [reply]
Thank you. It works on a single image but I can't figure out how to do it in batch.

I tried this: 
`For %A in (*.png) Do ( magick.exe %A ^
( +clone AlphaMask.png -alpha extract -compose multiply -composite ) ^
-alpha off -compose copy_opacity -composite ^
Output/%A )`

I got this error: 
`For %A in (*.png) Do ( magick.exe %A ^
More? ( +clone AlphaMask.png -alpha extract -compose multiply -composite ) ^
More? -alpha off -compose copy_opacity -composite ^
-alpha was unexpected at this time.`

What am I doing wrong?

**Comment by snibgo** [comment]
You need to escape the nested parentheses with a caret ^, or quote them.
```
For %A in (*.png) Do ( magick.exe %A ^
^( +clone hexagon.png -alpha extract -compose multiply -composite ^) ^
-alpha off -compose copy_opacity -composite ^
Output/%A )
```

**Reply by Isak-Andersson** [reply]
This gives me:
`(magick.exe ZSP3A0.png ^( +clone AlphaMask.png -alpha extract -compose multiply -composite ) -alpha off -compose copy_opacity -composite Output/ZSP3A0.png  )
magick.exe: unable to open image '^(': No such file or directory @ error/blob.c/OpenBlob/3565.
magick.exe: no decode delegate for this image format `' @ error/constitute.c/ReadImage/741.`

**Comment by snibgo** [comment, marked as answer]
Why do you have the outermost parentheses? I suggest you remove them. Then you don't need to escape the inner parentheses.
```
For %A in (*.png) Do %IMG7%magick %A ^
( +clone hexagon.png -alpha extract -compose multiply -composite ) ^
-alpha off -compose copy_opacity -composite ^
Output/%A 
```

**Comment by Isak-Andersson** [comment]
Aha. I thought you needed the parentheses. Now I know. Thank you very much.

What is %IMG7%, tho? This gives me:
`'%IMG7%magick' is not recognized as an internal or external command,`
I removed it and now it works.

**Comment by snibgo** [comment]
Yeah, sorry, my fault. Remove %IMG7%.

The directory for `magick` isn't on my system path, because I use many versions. So the commands on my computer use %IMG7% which is a Windows environment variable that points to whichever directory I need.
